lib/ansible/modules/cloud/amazon/redshift.py

```diff
@@ -243,11 +243,14 @@ def _collect_facts(resource):
         'db_name'           : resource['DBName'],
         'availability_zone' : resource['AvailabilityZone'],
         'maintenance_window': resource['PreferredMaintenanceWindow'],
+        'url'               : resource['Endpoint']['Address'],
+        'port'              : resource['Endpoint']['Port']
     }
 
     for node in resource['ClusterNodes']:
         if node['NodeRole'] in ('SHARED', 'LEADER'):
             facts['private_ip_address'] = node['PrivateIPAddress']
+            facts['public_ip_address'] = node['PublicIPAddress']
```

Comment:
Do clusters created without publicly_accessible being set have this key? (just wondering whether it should be `node.get('PublicIPAddress')`)

Reply:
yes - I ran it with publicly_accessible `yes`, `no`, and undefined, and the node always has a `PublicIPAddress` key. Interestingly, even if the cluster is not publicly accessible, AWS assigns it a public IP address value - not sure why.